lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -236,22 +467,48 @@ if ([Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
 }
 [Net.ServicePointManager]::SecurityProtocol = $security_protocols
 
+# Check for case $checksum variable contain url. If yes, get file data from url and replace original value in $checksum
+if ($checksum_url) {
+
+    if ($checksum_url.startswith('http://', 1) -or $checksum_url.startswith('https://', 1) -or `
+        $checksum_url.startswith('ftp://', 1) -or [bool]([System.Uri]$checksum_url).isFile) {
+
+        $checksum = Get-ChecksumFromUrl -module $module -url $checksum_url -credentials $credentials `
```

Comment:
I seems like we have a few common parameters used in these cmdlets. It might be wise to create a common hashtable that contains the common args and splat them when calling these cmdlets like;

```
$common_args = @{
    module = $module
    credentials = $credentials
    headers = $headers
    .... # and so on
}
Get-ChecksumFromUrl @common_args
```

Reply:
Good hint, thank you!